### PR TITLE
Openssl

### PR DESCRIPTION
**Pull request by riataman**

### CODE REVIEW

**Comment by riataman** [comment]
Initial OpenSSL tracing. Dumps the secrets in a wireshark friendly format. Also dumps data in SSL_write and SSL_read. Next it would interesting to do redirecting in the functions to validate certificates.

**Comment by erikbor** [comment]
Nice one Pablo! This is a really exciting feature

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ env:
     - CMOCKA_INSTALL="$HOME/builds/cmocka-install"
     - LOGNAME="Travis"
 
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update          ; fi
```
Comment (efiop) [inline .travis.yml]: No trailing white spaces please.

File: ssl.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		trace_printf(0, " ");
+		print_key(ssl->session->master_key, ssl->session->master_key_length);
+		trace_printf(0, "\n");
+	}
+}
+
+
+int RETRACE_IMPLEMENTATION(SSL_write)(SSL *ssl, const void *buf, int num)
+{
+	rtr_SSL_write_t real_SSL_write = RETRACE_GET_REAL(SSL_write);
```
Comment (efiop) [inline ssl.c]: No mixed declarations and code please.

**Pull request merged**